bus.py

```diff
@@ -0,0 +1,97 @@
+import random
+from utility import distance, firstn, position_on_edge
+
+
+def _road_point_generator(roads):
+    assert len(roads) > 0
+
+    # Length of all roads combined
+    total_length = sum(map(lambda road: road.getLength(), roads))
+
+    while True:
+        # Select a point on the combined stretch of road
+        distance = random.uniform(0, total_length)
+
+        # Find the selected road
+        length_sum = 0.0
+        found_road = False
+        for road in roads:
+            from_node = road.getFromNode().getCoord()
+            to_node = road.getToNode().getCoord()
+
+            if (length_sum + road.getLength()) >= distance:
+                # Compute the exact point on the selected road
+
+                # Distance along the road segment
+                remaining = length_sum - distance
+                pos = position_on_edge(road, remaining)
+
+                yield [
+                        pos[0],
+                        pos[1],
+                        road,
+                        remaining]
+                found_road = True
+                break
+            else:
+                length_sum += road.getLength()
+
+        if not found_road:
+            raise AssertionError("Failed to pick a road. A distance beyound the last road must have been erroneously picked: {} (length sum: {}) (total length: {})".format(distance, length_sum, total_length))
+
+
+def bus_stop_generator(roads, inner_r, outer_r, k=10, seeds=[]):
```

Comment:
Never use mutable defaults arguments. See https://docs.python-guide.org/writing/gotchas/

Reply:
fixed in 77cbece